sdk/README.md

```diff
@@ -1,25 +1,29 @@
 # Lookout Analyzer SDK
 
-An analyzer is a gRPC server, that implements [Analyzer service](https://github.com/src-d/lookout/tree/master/sdk/service_analyzer.proto) to receive events from the server.
+An analyzer is a [gRPC](https://grpc.io/) server that implements the [protocol buffers Analyzer service](./service_analyzer.proto) to receive events from the lookout server.
 
-An analyzer should use gRPC client to access [Data service](https://github.com/src-d/lookout/tree/master/sdk/service_data.proto) to get content or UAST of the changes.
+At the same time, an analyzer will act as a gRPC client to access the [Data Server](./service_data.proto) exposed by the lookout server. This endpoint will provide the necessary source code and [Babelfish Universal Abstract Syntax Trees (UAST)](https://doc.bblf.sh/uast/uast-specification.html).
 
-All `.proto` files are located in [/sdk](https://github.com/src-d/lookout/tree/master/sdk) directory.
+All the needed `.proto` files are located in the current [/sdk directory](./).
 
 
 ## Code generation
 
+You can create a new analyzer in any language that supports protocol buffers.
+Please refer to the [official Protocol Buffers documentation](https://developers.google.com/protocol-buffers/) to learn how to get started.
+
 ### Python
 
-Dependencies:
+For convenience we are including here a simplified example on how to generate Python code.
+You have more detailed instructions in the [official gRPC Python Quickstart guide](https://grpc.io/docs/quickstart/python.html).
+
+First install the dependencies:
 
 ```bash
 $ pip install grpcio-tools
 ```
 
-Read more about gRPC in [Python Quickstart](https://grpc.io/docs/quickstart/python.html).
-
-Generation:
+Now clone this repository and run this command from the root directory:
```

Comment:
This example actually does not work because we need to generate two dependencies - without them nothing works. See https://github.com/src-d/style-analyzer/blob/master/lookout/core/generate_rpc.sh#L10

Reply:
Thank you for pointing this out!

Suggestion: change to

```
python -m grpc_tools.protoc -Isdk \
     --python_out=./sdk/out --grpc_python_out=./sdk/out \
     $(find sdk -name *.proto)
```
and add `find sdk/out -iname gogo_pb2_grpc1.py | grep ".*"` to the code generation CI.

Reply:
:+1: I added `ls "$PY_OUT_DIR"/*.py` instead, which will fail if there isn't any `.py`, and should help us to keep an eye on what files were generated.

Reply:
Fellows, I have recently found a very nasty problem with this. It conflicts with `bblfsh` Python package: whichever is imported first wins, the second guy explodes :) The reason is because the protobuf descriptors have global names and `bblfsh` uses the same dependencies. I had to hack to fix it: https://github.com/src-d/style-analyzer/blob/master/lookout/core/generate_rpc.sh#L12

But anyway this example will work in isolation from the other packages.

Reply:
@carlosms I didn't get how `ls` helps here. Current command generates files, but it has the line:
```
from github.com.gogo.protobuf.gogoproto import gogo_pb2 as github_dot_com_dot_gogo_dot_protobuf_dot_gogoproto_dot_gogo__pb2
```
which would fail in runtime.

we need to add generation for gogo or use similar bblfsh hack as Vadim described.

Reply:
Eh, @carlosms indeed the purpose of this discussion was different :) 

It was not about adding a check, but rather about to make sure we generate code for gogo things that was not generated before  (see `find` above).

There must be a better way of handling conflict that @vmarkovtsev found, but that should be discussed/coordinated with LA in a separate issue.

For now, let's include a gogo generation in SDK doc example (and add notice about possible runtime conflict \w bblfsh + a link to workaround from above).

I think it's OK to keep ls for debugging, at the same time - let's add a check specifically to make sure gogo things are generated as well.

Reply:
Got it. Done in 67e44ed.